src/IPStack/Packet/PacketProtocolParser.swift

```diff
@@ -62,10 +62,11 @@ class UDPProtocolParser: TransportProtocolParserProtocol {
             packetData.replaceSubrange(offset+4..<offset+6, with: $0)
         }
         packetData.replaceSubrange(offset+8..<offset+8+payload.count, with: payload)
-        packetData.resetBytes(in: offset+6..<offset+8)
 
-        // Todo: compute checksum
-//        var checksum = Checksum.computeChecksum(datagram, from: 0, to: nil, withPseudoHeaderChecksum: pseudoHeaderChecksum)
-//        datagram.replaceBytesInRange(NSRange(location: 6, length: 2), withBytes: &checksum)
+        packetData.resetBytes(in: offset+6..<offset+8)
+        var checksum = Checksum.computeChecksum(packetData, from: 0, to: nil, withPseudoHeaderChecksum: pseudoHeaderChecksum)
```

Comment:
Yes, it's OK. Perhaps because `computePseudoHeaderChecksum()`  function returns the value is wrong. 
![udpchecksum](https://cloud.githubusercontent.com/assets/548532/23108955/410554a4-f74f-11e6-90d5-9e307a2f1c36.png)